src/main/java/eu/bbmri_eric/negotiator/service/DBAttachmentService.java

```diff
@@ -98,103 +140,71 @@ private AttachmentMetadataDTO saveAttachment(
     }
   }
 
-  private void checkAuthorization(Long userId, Negotiation negotiation) {
-    Person uploader =
-        personRepository.findById(userId).orElseThrow(() -> new EntityNotFoundException(userId));
-    if (!uploader.isAdmin()
-        && !uploader.equals(negotiation.getCreatedBy())
-        && negotiation.getResources().stream().noneMatch(uploader.getResources()::contains)) {
+  private void checkAuthorization(Long userId, String negotiationId, String organizationExternalId)
+      throws ForbiddenRequestException {
+    // if the recipient organization is specified and the sender is not the negotiation creator,
+    // checks whether the user is representative of the organization
```

Comment:
please remove comments and rather extract the logic into a separate method with an appropriate name